src/engraving/rendering/score/tlayout.cpp

```diff
@@ -6136,6 +6140,7 @@ void TLayout::layoutTimeSig(const TimeSig* item, TimeSig::LayoutData* ldata, con
     int numOfLines = 0;
     TimeSigType sigType = item->timeSigType();
     const Staff* staff = item->staff();
+    const Fraction tick = item->isCourtesy() ? item->tick() - Fraction::eps() : item->tick();
```

Comment:
Also, as far as I understand, you are using isCourtesy just to determine that this timeSig is at the end of the previous measure, rather than at the beginning of the next (meaning that it should use the previous values of staff->lines and staff->lineDist). For this, perhaps you could simply do staff->lines(item->segment()->measure()->tick()) (this will automatically pick the staffline from the previous measure if this is a courtesy